forest/tree/tree.go

```diff
@@ -0,0 +1,58 @@
+package tree
+
+import (
+	cache "github.com/treeverse/lakefs/forest/cache_map"
+	gr "github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/committed/sstable"
+)
+
+var minimalKey = gr.Key(nil)
+
+type treePartType struct {
+	PartName sstable.ID `json:"part_name"`
+	MaxKey   gr.Key     `json:"max_path"`
+}
+type TreeType struct {
+	treeSlice []treePartType
+}
+
+type treesRepo struct {
+	treesMap   cache.CacheMap
+	partManger sstable.Manager
+}
+
+const (
+	CacheMapSize     = 1000
+	CacheTrimSize    = 100
+	InitialWeight    = 64
+	AdditionalWeight = 16
+	TrimFactor       = 1
+)
+
+func InitTreesRepository(manager sstable.Manager) TreeRepo {
+	treesRepository := &treesRepo{
+		treesMap:   cache.NewCacheMap(CacheMapSize, CacheTrimSize, InitialWeight, AdditionalWeight, TrimFactor),
+		partManger: manager,
+	}
+	return treesRepository
+}
+
+func (t *treesRepo) GetPartManger() sstable.Manager {
+	return t.partManger
+}
+
+type TreeRepo interface {
+	NewTreeWriter(splitFactor int, closeAsync sstable.BatchWriterCloser) TreeWriter
+	NewScannerFromID(treeID gr.TreeID, start gr.Key) (gr.ValueIterator, error)
+	NewScannerFromTreeParts(treeSlice TreeType, start gr.Key) (gr.ValueIterator, error)
+	GetPartManger() sstable.Manager
```

Comment:
Does the tree interface must expose the inner part manager? It's the sort of complication we really need to avoid if possible.

Reply:
Currently process initialization, and where application parts get trees and sstables is not finalized AFAIK. Once it is - I will happily remove it.

Reply:
Why is initialisation part of the interface?
Isn't it a one time operation during setup?